Python/import.c

```diff
@@ -1744,10 +1751,461 @@ import_find_and_load(PyThreadState *tstate, PyObject *abs_name)
     return mod;
 }
 
+static int
+lazy_loaded_contains_parent(PyObject *module, PyObject *name)
+{
+    assert(module != NULL);
+    PyObject *lazy_submodules = PyObject_GetAttr(module, &_Py_ID(__lazy_submodules__));
+    if (lazy_submodules == NULL) {
+        PyErr_Clear();
+        return 0;
+    }
+    int res = PySet_Contains(lazy_submodules, name);
+    Py_DECREF(lazy_submodules);
+    if (res < 0) {
+        return -1;
+    }
+    return res;
+}
+
+static int
+lazy_loaded_add_parent(PyObject *module, PyObject *name)
+{
+    assert(module != NULL);
+    PyObject *lazy_submodules = PyObject_GetAttr(module, &_Py_ID(__lazy_submodules__));
+    if (lazy_submodules == NULL) {
+        PyErr_Clear();
+        lazy_submodules = PySet_New(NULL);
+        if (lazy_submodules == NULL) {
+            return -1;
+        }
+        if (PyObject_SetAttr(module, &_Py_ID(__lazy_submodules__), lazy_submodules) < 0) {
+            Py_DECREF(lazy_submodules);
+            return -1;
+        }
+    }
+    int res = PySet_Add(lazy_submodules, name);
+    Py_DECREF(lazy_submodules);
+    if (res < 0) {
+        return -1;
+    }
+    return 0;
+}
+
+static PyLazyImportObject *
+new_lazy_import(PyObject *parent, PyObject *child, PyObject *globals, PyObject *locals)
+{
+    PyObject *fromlist = PyTuple_New(1);
+    if (fromlist == NULL) {
+        return NULL;
+    }
+    Py_INCREF(child);
+    if (PyTuple_SetItem(fromlist, 0, child) < 0) {
+        return NULL;
+    }
+    PyObject *from = _PyLazyImport_NewModule(parent, globals, locals, fromlist, NULL);
+    Py_DECREF(fromlist);
+    if (from == NULL) {
+        return NULL;
+    }
+    PyLazyImportObject *lazy_import = (PyLazyImportObject *)_PyLazyImport_NewObject(from, child);
+    Py_DECREF(from);
+    return lazy_import;
+}
+
+static int
+feed_lazy_loaded(PyThreadState *tstate, PyObject *name)
```

Comment:
this name is definitely confusing, but I don't understand the purpose of the function well enough yet to suggest a different name :)

Reply:
this is different from `__lazy_submodules__ ` in a module in that this is for lazy submodules of modules that don't yet exist. For example, with `import foo.bar.baz; import foo.ack`, after `feed_lazy_loaded()`, `lazy_attributes` would end up with something like:

```py
lazy_attributes == {
    "foo": {"bar", "ack"},
    "foo.bar": "baz",
}
```

So that when you ultimately do `import foo`, it'd add lazy attributes `bar` and `ack` to it and also discard the module from `lazy_attributes`.